kamu-core/src/domain/repos/metadata_chain.rs

```diff
@@ -327,10 +332,32 @@ impl From<super::reference_repository::SetRefError> for AppendError {
     }
 }
 
+impl From<GetBlockError> for AppendError {
+    fn from(v: GetBlockError) -> Self {
+        match v {
+            GetBlockError::NotFound(e) => {
+                AppendError::InvalidBlock(AppendValidationError::PrevBlockNotFound(e))
```

Comment:
I've inlined the conversion here and in `validate_append_system_time_is_monotonic`, which seemed identical.
I understand the point about automatic error conversions not being a silver bullet.
But I do not understand the problem you've described about  `n-(2..n)`.

Reply:
By `n - (2..n)` I meant the situation when appending block `n` we encounter a missing block in the chain other than the current `head` (`n - 1`).

Crude example:
- Imagine we're appending `SetTransform` metadata event
- Since this event only makes sense for `Derivative` datasets we decide to write such validator
- Validator has to walk the entire chain back to the `Seed` block to understand the type of the dataset
- Person implementing such iteration will be tempted to deal with errors using `?` operator, e.g. `self.get_block(curr.prev_block_hash)?`
- If the chain is malformed and missing some block this will trigger the automatic conversion from `NotFound -> PrevBlockNotFound` i.e. we will get a likely very confusing error that is not indicative of the real problem

Reply:
Okay, so by getting rid of that automatic error mapping I've already prevented this scenario.